e2e/cypress/support/api_commands.js

```diff
@@ -387,6 +387,121 @@ Cypress.Commands.add('apiLinkGroupChannel', (groupID, channelID) => {
+Cypress.Commands.add('apiGetCustomRetentionPolicy', (id) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: `/api/v4/data_retention/policies/${id}`,
+        method: 'GET',
+        timeout: 60000,
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response);
+    });
+});
+
+/**
+ * Delete Custom Retention Policy
+ * @param {string} id - The id of the policy
+ */
+Cypress.Commands.add('apiDeleteCustomRetentionPolicy', (id) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: `/api/v4/data_retention/policies/${id}`,
+        method: 'DELETE',
+        timeout: 60000,
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response);
+    });
+});
+
+/**
+ * Get Custom Retention Policy teams
+ * @param {string} id - The id of the policy
+ */
+Cypress.Commands.add('apiGetCustomRetentionPolicyTeams', (id, page = 0, perPage = 100) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: `/api/v4/data_retention/policies/${id}/teams?page=${page}&per_page=${perPage}`,
+        method: 'GET',
+        timeout: 60000,
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response);
+    });
+});
+
+/**
+ * Get Custom Retention Policy channels
+ * @param {string} id - The id of the policy
+ */
+Cypress.Commands.add('apiGetCustomRetentionPolicyChannels', (id, page = 0, perPage = 100) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: `/api/v4/data_retention/policies/${id}/channels?page=${page}&per_page=${perPage}`,
+        method: 'GET',
+        timeout: 60000,
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response);
+    });
+});
+
+/**
+ * Search Custom Retention Policy teams
+ * @param {string} id - The id of the policy
+ */
+Cypress.Commands.add('apiSearchCustomRetentionPolicyTeams', (id, term) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: `/api/v4/data_retention/policies/${id}/teams/search`,
+        method: 'POST',
+        timeout: 60000,
+        body: {term},
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response);
+    });
+});
+
+/**
+ * Search Custom Retention Policy teams
+ * @param {string} id - The id of the policy
+ */
+Cypress.Commands.add('apiSearchCustomRetentionPolicyChannels', (id, term) => {
+    return cy.request({
+        headers: {'X-Requested-With': 'XMLHttpRequest'},
+        url: `/api/v4/data_retention/policies/${id}/channels/search`,
+        method: 'POST',
+        timeout: 60000,
+        body: {term},
+    }).then((response) => {
+        expect(response.status).to.equal(200);
+        return cy.wrap(response);
+    });
+});
```

Comment:
We're no longer adding new API commands in this file as we're organizing it at `/support/api/`. With that please move to that folder and create a new file as `/support/api/data_retention.js`. Also, please add corresponding types at ``/support/api/data_retention.d.ts``

Reply:
Per our DM conversation, cypress tests are still using the old matermost-redux repo so data retention types are not available. When cypress tests are updated, I can move these into a typed file.